File: anblog/src/app/models/category.ts
import { ObjectId } from './objectid';

class SubCategory {
  _id: ObjectId;
  name: string;
}

export class Category {
  _id: ObjectId;
  name: string;
  father: ObjectId;
  type: number;
  url: string;
  level: number;
  children: Array<string>;
  showFlag: number;
  subCategories: Array<Category>;
}

File: blogan/models/category.py
from blogan.extensions import db

class Category(db.Document):
    name = db.StringField(required=True)
    father = db.ReferenceField('self')
    type = db.IntField(default=0) #0-normal 1-chapter
    url = db.StringField(required=True, unique=True)
    level = db.IntField(default=1) #0-root 1-level 1
    children = db.ListField(db.StringField())
    showFlag = db.IntField(default=1) #0-hidden 1-show